src/config.rs

```diff
@@ -334,6 +334,90 @@ impl Context {
     }
 }
 
+// Separate impl block for self address handling
+impl Context {
+    /// determine whether the specified addr maps to the/a self addr
+    pub(crate) async fn is_self_addr(&self, addr: &str) -> Result<bool> {
+        let addr = addr_normalize(addr).to_lowercase();
+
+        // The addresses we get here are already normalized and lowercase
+        Ok(
+            self.get_primary_self_addr().await.ok().as_deref() == Some(&addr)
```

Comment:
This silently ignores database errors in `get_primary_self_addr`, see my comment on `get_primary_self_addr`.